library/common/extensions/filters/http/platform_bridge/filter.cc

```diff
@@ -133,24 +145,32 @@ Http::FilterDataStatus PlatformBridgeFilter::onData(Buffer::Instance& data, bool
       internal_buffer->drain(internal_buffer->length());
     }
     iteration_state_ = IterationState::Stopped;
-    break;
-  default:
-    PANIC("unsupported status for platform filters");
-  }
+    return Http::FilterDataStatus::StopIterationNoBuffer;
 
-  // TODO(goaway): Current platform implementations expose immutable data, thus any modification
-  // necessitates a full copy. Add 'modified' bit to determine when we can elide the copy. See also
-  // https://github.com/lyft/envoy-mobile/issues/949 for potential future optimization.
-  if (iteration_state_ == IterationState::Ongoing) {
-    data.drain(data.length());
-    data.addBufferFragment(*Buffer::BridgeFragment::createBridgeFragment(result.data));
+  case kEnvoyFilterDataStatusResumeIteration:
+    if (iteration_state_ != IterationState::Stopped) {
+      PANIC("invalid filter state: ResumeIteration  may only be used when filter iteration is "
+            "stopped");
+    }
+    if (result.extra_headers) {
+      PlatformBridgeFilter::replaceHeaders(**pending_headers, *result.extra_headers);
+      *pending_headers = nullptr;
+      free(result.extra_headers);
+    }
+    internal_buffer->drain(internal_buffer->length());
+    internal_buffer->addBufferFragment(*Buffer::BridgeFragment::createBridgeFragment(result.data));
+    return Http::FilterDataStatus::Continue;
+
+  default:
```

Comment:
I think it's worth having this be exhaustive so that if upstream Envoy adds a new iteration state Envoy Mobile won't compile

Reply:
Unfortunately we won't be able to get these compile-time guarantees here regardless. C/C++ don't enforce this, and static analysis would miss it too, since we're iterating over a constant set only partially defined in terms of the internal enum (ResumeIteration doesn't map to anything internal).